### PR TITLE
virtualmachine: Add support for Disk Encryption Set parameter

### LINKED ISSUE #425: Ansible support enable VM Disk Encryption
##### SUMMARY
Ansible support for Azure Disk Encryption for virtual machines and virtual machine scale sets.

https://docs.microsoft.com/en-us/cli/azure/vm/encryption?view=azure-cli-latest

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
Most probably a new Ansible module

https://docs.microsoft.com/en-us/cli/azure/vm/encryption?view=azure-cli-latest
https://docs.microsoft.com/en-us/azure/virtual-machines/linux/disk-encryption-cli-quickstart

### PR DESCRIPTION
**Pull request by lm-sig**
Fixes https://github.com/ansible-collections/azure/issues/425

Supersedes https://github.com/ansible-collections/azure/pull/1217

##### SUMMARY
I had a requirement to have virtual machines with disks (both OS and data) that used encryption. The VM module did not support provisioning or attaching any disk type with encryption enabled.

Fixes #425

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

azure_vm_virtualmachine

##### ADDITIONAL INFORMATION
A customer may set an Azure policy that requires VMs to have disks with encryption. If you attempt to provision a virtual machine with disks without encryption your request will fail.

This change will support both encrypted or unencrypted disks. Currently it requires you to provide the full Azure resource ID for the data encryption set ID.

### CODE REVIEW

**Comment by lm-sig** [comment]
There are currently (2023-11-01) no merge conflicts and no outstanding review issues. Please review as soon possible. Thanks.

File: plugins/modules/azure_rm_virtualmachine.py [hunk 1]
```diff
@@ -306,6 +315,10 @@
                 choices:
                     - ReadOnly
                     - ReadWrite
+                default: ReadOnly
+            disk_encryption_set_id:
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_virtualmachine.py:319]: Missing type and argument 'disk_encryption_set_id' found in data_disks is listed in DOCUMENTATION.options, but not accepted by the module argument_spec

File: plugins/modules/azure_rm_virtualmachine.py [hunk 2]
```diff
@@ -306,6 +315,10 @@
                 choices:
                     - ReadOnly
                     - ReadWrite
+                default: ReadOnly
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_virtualmachine.py]: Argument 'caching' in argument_spec found in data_disks defines default as (None) but documentation defines default as ('ReadOnly')

**Review by Fred-sun — COMMENTED** [review]
Please add test cases for the newly added parameters, thank you!

File: plugins/modules/azure_rm_virtualmachine.py (hunk 1 above)
Comment (lm-sig) [reply plugins/modules/azure_rm_virtualmachine.py:319]: Thank you. I have addressed this. New commit has been pushed.

File: plugins/modules/azure_rm_virtualmachine.py (hunk 2 above)
Comment (lm-sig) [reply plugins/modules/azure_rm_virtualmachine.py]: Thank you. This was a mistake. I have addressed this. New commit has been pushed.

**Comment by lm-sig** [comment]
I added a test case and updated to the latest 'dev' branch changes. As of today (2023-11-13) there should be no outstanding issues. Thanks.

**Comment by Fred-sun** [comment]
@lm-sig Thanks for your update! The 'os_disk_encryption_set_id' was added in #1306 and 'disk_encrytpion_set_id' was added in PR #1309. Thank you!

**Comment by lm-sig** [comment]
It's unfortunate Red Hat / @ephracis didn't see this was being worked on for several months, but thank you for the update. I guess this can be closed. Is there an ETA on the 1.19.0 release?

**Pull request closed without merging**